server/methods/cart.js

```diff
@@ -0,0 +1,149 @@
+function verifyCorrectProduct(variantOptions, selectedVariantId) {
```

Comment:
this is performing a double check that variant is in productId, since Arrays from mongo don't always persist order.